stout/filter.h

```diff
@@ -44,10 +44,18 @@ struct _Filter {
       k_.Ended();
     }
 
+    void Register(Interrupt& interrupt) {
+      assert(interrupt_ == nullptr);
+      interrupt_ = &interrupt;
```

Comment:
It doesn't look like you're ever using `interrupt_`. Did you mean to? If not, all you need here is `k_.Register(interrupt);`.